File: dataStructures/graphs/graph.js
class Graph {
    
    constructor() {
        // store the nodes and edges in two separate arrays
        this.adjancencyList = new Map();
    }
    addVertex(vertex) {
        if (!this.adjancencyList.has(vertex)) this.adjancencyList.set(vertex, []);
    }
    addEdge(vertex1, vertex2) {
        // if either vertex doesn't exist, add it to the graph first    
        this.adjancencyList.get(vertex1).push(vertex2);
        this.adjancencyList.get(vertex2).push(vertex1);
    }
    removeEdge(vertex1, vertex2) {
        this.adjancencyList.get(vertex1).splice(this.adjancencyList.get(vertex1).indexOf(vertex2), 1);
        this.adjancencyList.get(vertex2).splice(this.adjancencyList.get(vertex2).indexOf(vertex1), 1); 
      
    }
    removeVertex(vertex) {
        while (this.adjancencyList.size > 0) {
            const adjacentVertex = this.adjancencyList.get(vertex).pop();
            this.removeEdge(vertex, adjacentVertex);
        }
        this.adjancencyList.delete(vertex);
    }
}

const g = new Graph();
g.addVertex('Tokyo');
g.addVertex('London');
g.addVertex('Rome');
g.addVertex('Paris');
g.addVertex('Berlin');
g.addVertex('New York');
g.addEdge('Tokyo', 'London');
g.addEdge('Tokyo', 'Rome');
g.addEdge('London', 'Paris');
g.addEdge('Paris', 'Berlin');
g.addEdge('Berlin', 'New York');
g.addEdge('Rome', 'New York');
g.removeEdge('Rome', 'New York')

console.log(g);

File: dataStructures/graphs/graph.py
class Graph:
    def __init__(self):
        self.nodes = {}

    def  add_node(self, node):
        self.nodes[node] = []
    def add_edge(self, node1, node2):
        self.nodes[node1].append(node2)
        self.nodes[node2].append(node1)
    def remove_edge(self, node1, node2):
        self.nodes[node1].remove(node2)
        self.nodes[node2].remove(node1)
    def remove_node(self, node):
        for n in self.nodes[node]:
            self.remove_edge(node, n)
        del self.nodes[node]
    def get_neighbors(self, node):
        return self.nodes[node]
    

g = Graph()
g.add_node('a')
g.add_node('b')
g.add_node('c')
g.add_node('d')
g.add_node('e')
g.add_edge('a', 'b')
g.add_edge('a', 'c')
g.add_edge('b', 'd')
g.add_edge('c', 'd')
g.remove_node('c')
print(g.get_neighbors('a'))

n = g.nodes
print(n)